pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesServiceTest.java

```diff
@@ -465,4 +467,19 @@ public void testWriterCache() throws Exception {
         admin.namespaces().deleteNamespace(NAMESPACE4);
         Assert.assertNull(service.getWriterCaches().synchronous().getIfPresent(NamespaceName.get(NAMESPACE4)));
     }
+
+    @Test
+    public void testPrepareInitPoliciesCacheAsyncWhenNamespaceBeingDeleted() throws Exception {
+        SystemTopicBasedTopicPoliciesService service = (SystemTopicBasedTopicPoliciesService) pulsar.getTopicPoliciesService();
+        admin.namespaces().createNamespace(NAMESPACE5);
+
+        pulsar.getPulsarResources().getNamespaceResources().setPolicies(NamespaceName.get(NAMESPACE5),
+                old -> {
+                    old.deleted = true;
+                    return old;
+                });
+
+        service.prepareInitPoliciesCacheAsync(NamespaceName.get(NAMESPACE5)).get();
```

Comment:
Could you check `policyCacheInitMap` keys does not include the `NAMESPACE5`?

Reply:
Because I used a new namespace here, it does not exist in policyCacheInitMap

Reply:
I know, just to assert this case.

Reply:
I get, I will add a assert for this case.